### PR TITLE
[datadog_powerpack] add support for datadog_powerpack data source

### PR DESCRIPTION
**Pull request by bharling**
Just based off the dashboard data source, with basic matching on [Datadog Powerpack](https://www.datadoghq.com/blog/standardize-dashboards-powerpacks-datadog/) resources by name.

### CODE REVIEW

**Comment by drichards-87** [comment]
@bharling Thanks for submitting this PR. I'll let the API Clients team handle the tech review. For the review of the `.md` content, I've created a Jira card for the Docs Team. We'll get someone assigned to the ticket this week.

**Comment by nkzou** [comment]
Hi, we aren't allowing new SDKv2 resources in our terraform provider anymore. If you could convert this to a `terraform-plugin-framework` resource that would be great, otherwise we'll try to get to this soon.

**Comment by bharling** [comment]
> Hi, we aren't allowing new SDKv2 resources in our terraform provider anymore. If you could convert this to a `terraform-plugin-framework` resource that would be great, otherwise we'll try to get to this soon.

Got it, no worries I'll refactor and update this PR

**Comment by bharling** [comment]
@nkzou - I've refactored this to use the Framework provider instead of legacy - tests appear to be passing now and docs are updated, would you be able to take a look?

**Review by nkzou — DISMISSED** [review]

**Review by dashashifrina — DISMISSED** [review]
Thanks for contributing!

File: docs/data-sources/powerpack.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+Use this data source to retrieve information about an existing Datadog Powerpack.
+
+
+
+<!-- schema generated by tfplugindocs -->
+## Schema
+
+### Required
+
+- `name` (String) The name of the powerpack to search for.
```
Comment (buraizu) [inline docs/data-sources/powerpack.md:20]:
```suggestion
- `name` (String) The name of the Powerpack to search for.
```

**Review by buraizu — APPROVED** [review]
Thanks, approving with a minor update requested

**Review by nkzou — APPROVED** [review]

**Pull request merged**